fix: line-clamp property does not truncate long texts properly in app card description

Anshumancanrock opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

This PR fixes a critical cross-browser compatibility issue where app card descriptions were not properly truncated due to reliance on webkit-only CSS properties. Long descriptions were overflowing and breaking the layout in many non-webkit browsers.

- Fixes #23659 
- Fixes [CAL-6379](https://linear.app/calcom/issue/CAL-6379/bug-line-clamp-prperty-does-not-truncate-long-text-properly)

### Key improvements:

Replaces webkit-only -webkit-line-clamp with standard CSS properties that work across all browsers

Implements consistent 3-line truncation with JavaScript fallback for precise character limits

## Visual Demo (For contributors especially)

#### Image Demo (if applicable):

Before Fix:
<img width="1602" height="778" alt="Screenshot 2025-12-04 015138" src="https://github.com/user-attachments/assets/5550adad-f3ea-4675-8a07-324b39ae928e" />




After Fix:


https://github.com/user-attachments/assets/c332832a-0ad4-4797-afe0-e18a80b8ff99





## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- No special environment variables required
- Standard Cal.com development environment

Anshumancanrock commented:
@anikdhabal Hi, Coderabbit has suggested removing line-clamp across the whole repo and replacing it with my approach. Would you recommend that I go ahead with this change?

kart1ka commented on packages/features/apps/components/AppCard.tsx:
```diff
@@ -117,15 +118,14 @@ export function AppCard({ app, credentials, searchText, userAdminTeams }: AppCar
+          wordBreak: "break-word",
         }}>
-        {app.description}
+        {app.description ? (app.description.length > maxDescriptionLength ? `${app.description.substring(0, maxDescriptionLength).trim()}...` : app.description) : ""}
```
can we simplify this code here?

kart1ka reviewed (changes requested):
This does not work properly. There are cases where the text content is simply cutt off without showing "..."
See the video attached: https://cap.link/js0bvmqd57qqm4j

Anshumancanrock commented:
@dhairyashiil could you please review it ser? The video after fix is already there.

dhairyashiil reviewed (commented):
Hello @Anshumancanrock,
Could you please refer to #23660 and attach videos showing the different types of behaviors the maintainer and contributor discussed there? This would help me a lot in reviewing and approving the PR.

Also, please make sure to check and record the behavior across all sizes like mobile, iPad, laptop, etc.

Thanks a ton!

Anshumancanrock commented:
@dhairyashiil yeah ! Attached the video there and here too, showing fix in both web and mobile view.

https://github.com/user-attachments/assets/6a6faa01-17f8-4c0e-ad79-09213c918535

Anshumancanrock commented:
I fixed two issues:

1. The truncate ( … ) was not showing even where it was supposed to appear.

2. The text was getting cut off at the bottom of the app card.

I also added one extra improvement: truncating the title when it exceeds one line. Without this, some app discription could never truncate properly since we are using a line clamp-3 for the description.

Hope this helps @dhairyashiil

Anshumancanrock committed:
fix: implement cross-browser compatible text truncation in app cards

Anshumancanrock commented:
@anikdhabal the microsoft teams app card's description is still not truncating
<img width="397" height="330" alt="image" src="https://github.com/user-attachments/assets/bba7ad71-fae6-4af7-95b7-a17b31561555" />

Since the description is set to truncate after 3 lines and this particular card’s description only has 2 lines so truncation doesn’t take effect. If we truncate the long title to a single line, the description would extend to 3 lines and truncate properly.

Should I raise a PR to fix this specific case?
